### PR TITLE
Update MSFT_EXOOrganizationConfig.psm1

### PR DESCRIPTION
**Pull request by stvnmbr1**
Verify AutoExpandingArchiveEnabled is True in organisation before invoke

#### Pull Request (PR) description

catch error - when AutoExpandingArchiveEnabled is True already, deployment breaks
Verify AutoExpandingArchiveEnabled is True in organisation -> if true, delete AutoExpandingArchiveEnabled from $SetValues

#### This Pull Request (PR) fixes the following issues
None

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244)

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1 [hunk 1]
```diff
@@ -771,6 +771,14 @@ function Set-TargetResource
     $SetValues.Remove('CertificateThumbprint') | Out-Null
     $SetValues.Remove('CertificatePath') | Out-Null
     $SetValues.Remove('CertificatePassword') | Out-Null
+    
+    $isAutoExpandingArchiveEnabled = Get-OrganizationConfig | select AutoExpandingArchiveEnabled -ExpandProperty AutoExpandingArchiveEnabled
+
+    if ($isAutoExpandingArchiveEnabled -eq $True)
+    {
+        $SetValues.Remove('AutoExpandingArchiveEnabled') | Out-Null
```
Comment (stvnmbr1) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1:779]: AutoExpandingArchiveEnabled should be AutoExpandingArchive

**Comment by stvnmbr1** [comment]
added bugfix for TlsDomain in EXOOutboundConnector

Cannot process argument transformation on parameter 'TlsDomain'. Cannot convert the "System.Collections.ArrayList" value of type "System.Collections.ArrayList" to type "Microsoft.Exchange.Data.SmtpDomainWithSubdomains".

**Comment by NikCharlebois** [comment]
@stvnmbr1 Can you please make sure you merge the latest changes in your branch. Unit tests are failing. Thanks

**Comment by stvnmbr1** [comment]
not sure why they are failing? I'm able to run without issues. followed doc https://docs.microsoft.com/en-us/microsoft-365/compliance/enable-unlimited-archiving?view=o365-worldwide


`$isAutoExpandingArchiveEnabled = Get-OrganizationConfig | select AutoExpandingArchiveEnabled -ExpandProperty AutoExpandingArchiveEnabled`

returns true for me. Unit test doesn't find the AutoExpandingArchiveEnabled property

**Comment by NikCharlebois** [comment]
Looks like the Unit Tests are failing. Let us know if you need assistance in correcting the issue. Thanks

**Comment by stvnmbr1** [comment]
Hi @NikCharlebois 

i could use some help, as i can execute the command without any issue.

Unit test looks like it does not find the AutoExpandingArchiveEnabled property, after executing Get-OrganizationConfig.
This should not cause an issue as i was following microsoft docs, mentioned above.
Unit test logs don't show much info to debug the issue

thanks in advance!

**Comment by NikCharlebois** [comment]
I believe the fix here is fairly simple. Since the property is already a boolean, we don't need to call the ExpandProperty switch. Therefore the command would simply become

$isAutoExpandingArchiveEnabled = Get-OrganizationConfig | Select-Object AutoExpandingArchiveEnabled

**Review by NikCharlebois — CHANGES_REQUESTED** [review]
Many picky comments and one thing to address with the .schema.mof not matching the changes

> Reviewed 1 of 2 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-:-Mdvt5ptAJ7q7FSKE1r2:b-54kodu)* status: all files reviewed, 6 unresolved discussions (waiting on @stvnmbr1)

___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1, line 774 at r3](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MdvsRTPBD8inwBIvRHD:-MdvsRTPBD8inwBIvRHE:b-kdq1w4) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/8ed2e7b58f29e87c7cd759f476853ade0f3f6e8a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1#L774)):*
> ```PowerShell
>     $SetValues.Remove('CertificatePath') | Out-Null
>     $SetValues.Remove('CertificatePassword') | Out-Null
>     
> ```

Remove spaces on empty line
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1, line 775 at r3](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MdvsX5O-SxNbLC1ZkNi:-MdvsX5O-SxNbLC1ZkNj:bf009d9) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/8ed2e7b58f29e87c7cd759f476853ade0f3f6e8a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1#L775)):*
> ```PowerShell
>     $SetValues.Remove('CertificatePassword') | Out-Null
>     
>     $isAutoExpandingArchiveEnabled = Get-OrganizationConfig | select AutoExpandingArchiveEnabled
> ```

Please use the full Select-Object form for consistency
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1, line 776 at r3](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MdvsQUz1UbU5QpfuZk8:-MdvsQV-3-7NtlhtI-JA:b-kdq1w4) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/8ed2e7b58f29e87c7cd759f476853ade0f3f6e8a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1#L776)):*
> ```PowerShell
>     
>     $isAutoExpandingArchiveEnabled = Get-OrganizationConfig | select AutoExpandingArchiveEnabled
>     
> ```

Remove spaces on empty line
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1, line 781 at r3](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MdvsOm9CzVGxpdbWHbD:-MdvsOmA4zwL13bzkSwr:b-kdq1w4) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/8ed2e7b58f29e87c7cd759f476853ade0f3f6e8a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1#L781)):*
> ```PowerShell
>         $SetValues.Remove('AutoExpandingArchive') | Out-Null
>     } 
>     
> ```

Remove spaces on empty line
___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOutboundConnector/MSFT_EXOOutboundConnector.psm1, line 59 at r3](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MdvszqB8rHfm3yr0uBK:-MdvszqB8rHfm3yr0uBL:bhmqsgp) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/8ed2e7b58f29e87c7cd759f476853ade0f3f6e8a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOutboundConnector/MSFT_EXOOutboundConnector.psm1#L59)):*
> ```PowerShell
>         [Parameter()]
>         [System.String]
>         $TlsDomain,
> ```

The .schema.mof file will also need to be updated to reflect this type change.

**Comment by stvnmbr1** [comment]
Made sense, did not think about the expandingproperty.
Thanks ! Unit tests are successfull now

**Review by NikCharlebois — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-:-Mdw6F_pDpLc-AKTJ23B:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @stvnmbr1)

**Comment by stvnmbr1** [comment]
Corrected the files to your remarks, thanks for the help
LGTM!

**Review by NikCharlebois — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244)* status: all files reviewed, 2 unresolved discussions

___
*[Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1, line 779 at r1](https://reviewable.io/reviews/microsoft/microsoft365dsc/1244#-MamHKOvY_bRKAuFUf9U-r1-779:-MdwBVav3E7l3fCZaQGb:bb7f2mf) ([raw file](https://github.com/microsoft/microsoft365dsc/blob/bb262aa67fead597e158a8e2d28f496f4c4c0eee/Modules/Microsoft365DSC/DSCResources/MSFT_EXOOrganizationConfig/MSFT_EXOOrganizationConfig.psm1#L779)):*
<details><summary><i>Previously, stvnmbr1 wrote…</i></summary><blockquote>

AutoExpandingArchiveEnabled should be AutoExpandingArchive
</blockquote></details>

done

**Pull request merged**